spec/integration/composite_primary_key_spec.rb

```diff
@@ -3,26 +3,150 @@
 require "spec_helper"
 
 describe "composite_primary_key" do
-  context "without associations" do
-    with_model :CompositePrimaryKeyModel do
-      table primary_key: [:prefix, :postfix] do |t|
-        t.string :prefix
-        t.string :postfix
-        t.string :name
+  context 'without relations' do
+    with_model :Parent do
+      table primary_key: [:first_name, :last_name] do |t|
+        t.string :first_name
+        t.string :last_name
+        t.string :hobby
       end
 
       model do
         include PgSearch::Model
-        self.primary_key = [:prefix, :postfix]
-        pg_search_scope :search_name, against: :name
+        self.primary_key = [:first_name, :last_name]
+        pg_search_scope :search_hobby, against: :hobby
       end
     end
 
-    before { CompositePrimaryKeyModel.create!(id: ["prefix", "postfix"], name: "bar") }
-    let!(:record_1) { CompositePrimaryKeyModel.create!(id: ["prefix_2", "postfix_2"], name: "foo") }
+    before { Parent.create!(id: ["first_name", "last_name"], hobby: "golf") }
+    let!(:record_1) { Parent.create!(id: ["first_name_2", "last_name_2"], hobby: "basketball") }
 
     it "searches without any issues" do
-      expect(CompositePrimaryKeyModel.search_name("foo")).to eq([record_1])
+      expect(Parent.search_hobby("basketball")).to eq([record_1])
+    end
+  end
+
+  context "without composite_primary_key, searching against relation with a composite_primary_key" do
+    with_model :Parent do
+      table primary_key: [:first_name, :last_name] do |t|
+        t.string :first_name
+        t.string :last_name
+        t.string :hobby
+      end
+
+      model do
+        include PgSearch::Model
+        has_many :children
+        self.primary_key = [:first_name, :last_name]
+      end
+    end
+    
+    with_model :Child do
+      table do |t|
+        t.string :parent_first_name
+        t.string :parent_last_name
+      end
+
+      model do
+        include PgSearch::Model
+        belongs_to :parent, foreign_key: [:parent_first_name, :parent_last_name]
+
+        pg_search_scope :search_parent_hobby, associated_against: {
```

Comment:
It's great that you are testing `associated_against`, good catch!

Reply:
this opened a rabbit hole that I don't think I was ready for 😅